pkg/route/api/v1/conversion.go

```diff
@@ -13,6 +13,9 @@ func init() {
 			obj.To.Kind = "Service"
 		},
 		func(obj *TLSConfig) {
+			if len(obj.Termination) == 0 {
+				obj.Termination = TLSTerminationEdge
```

Comment:
I don't want to have different special paths.  This problem doesn't bother
me enough to justify having a split brain on validation.

On Mon, Dec 14, 2015 at 10:22 AM, David Eads <notifications@github.com>
wrote:

> In pkg/route/api/v1/conversion.go
> <https://github.com/openshift/origin/pull/6076#discussion_r47509346>:
>
> > @@ -13,6 +13,9 @@ func init() {
> >  			obj.To.Kind = "Service"
> >  		},
> >  		func(obj *TLSConfig) {
> > +			if len(obj.Termination) == 0 {
> > +				obj.Termination = TLSTerminationEdge
>
> removing the TLS data in conversion means we can't catch it in validation.
> making an object in this state fail validation would mean it couldn't be
> updated in etcd without setting the termination type or removing the
> cert/key data... what would that break?
>
> Right, had a conversation in irc. If we tighten validation, do no
> defaulting, and create a special path for validateUpdate for routes/status,
> nothing breaks, but @smarterclayton <https://github.com/smarterclayton>
> wasn't convinced.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6076/files#r47509346>.
>